SPARQLTransformer.py

```diff
@@ -297,7 +297,7 @@ def fit(k):
 
         # variable not in result, delete from
         if variable not in line:
-            instance.pop(k)
+            instance[k] = None
```

Comment:
Setting to `None` has a different effect than removing the element.
Why do you prefer this version?

Reply:
@pasqLisena I've made some new changes so that it works the same as in previous versions. The only difference is that when it is a list we set the value to `None` and if all values in a list are `None` we pop the list. I think this is the best solution. What do you think?

Reply:
Yeah applying the new behaviour only to the lists is consistent.
I merge your PR!

Reply:
Published new 2.3.0 version including this